### DISCUSSION TITLE
AMS with autoscaling spamming clusterdb.clusternode

### DISCUSSION THREAD

**Discussion by SebastienGautier** [ant-media/Ant-Media-Server#6570] [Q&A]
Hello,

We're using kubernetes to host AMS with autoscaling, after it running for some time the clusterdb.clusternode collection is getting filled with old nodes that will never appear again. Is there some sort of cleanup happening at some point ? If not is this a planned feature ? I'm afraid it might get slower over time
![image](https://github.com/user-attachments/assets/496ce828-1ad1-4358-b6aa-c4633a6d15c0)

Thanks

**Comment by muratugureminoglu** [comment]
Hi @SebastienGautier 

Thanks for your question. 

To better understand the issue and assist you more effectively, could you please share the reproduce scenario? How can I reproduce this on my side? Could you please share step by step ? 

Additionally, could you run the following command in MongoDB and share the result with me? Also, to verify, could you provide a screenshot of the cluster menu from the dashboard? (Please remember to remove any Public IPs if they are visible.)

```
db.clusternode.find()
```

Regards.

**Reply by SebastienGautier** [reply]
Hi @muratugureminoglu,

You can reproduce by creating an autoscaling AMS infrastructure en forcing it to scale up and down, all removed nodes will remain in the clusternode table forever

Here is a view of the cluster page:
![image](https://github.com/user-attachments/assets/3a2c7990-9670-4bbf-aaf4-5007406b27e8)

How can I share the db.clusternode.find() export privately ?

**Reply by muratugureminoglu** [reply]
Hi @SebastienGautier 

 Please e-mail to support[@]antmedia[.]io by mentioning me. Then I can check it.
